src/command_modules/azure-cli-iot/azure/cli/command_modules/iot/_params.py

```diff
@@ -37,6 +37,8 @@ def get_device_id_completion_list(prefix, action, parsed_args, **kwargs):  # pyl
 # Arguments for 'iot hub policy' group
 register_cli_argument('iot hub policy', 'policy_name', options_list=('--name', '-n'), id_part='child_name',
                       help='Shared access policy name.')
+register_cli_argument('iot hub policy', 'permissions', help='Permissions of shared access policy.',
+                      **enum_choice_list(AccessRights))
```

Comment:
I am taking the first approach you suggested. 
Not sure whether IoT service team will accept such request. Will ask them when convenient. For now adding some conversion logic as work around to make the command parameter right.